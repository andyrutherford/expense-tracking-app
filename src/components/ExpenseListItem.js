import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ( { description, category, amount, createdAt, id } ) => (
    <div>
        <Link 
            to={`edit/${id}`}
            ><h3>{description}</h3>
        </Link>
        {category && (<p>[{category}]</p>) }
        <p>
            {numeral(amount / 100).format('$0,0.00')}{' '}
            - 
            {' '}{moment(createdAt).format('DD MMMM YYYY')}
        </p>
    </div>
);

export default ExpenseListItem;

