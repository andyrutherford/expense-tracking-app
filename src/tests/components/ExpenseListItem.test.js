import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem with an expense', () => {
    const expenseItem = expenses[1];
    const wrapper = shallow(
        <ExpenseListItem 
            description={expenseItem.description}
            amount={expenseItem.amount}
            createdAt={expenseItem.createdAt}
            id={expenseItem.id} /> );
    expect(wrapper).toMatchSnapshot();
});