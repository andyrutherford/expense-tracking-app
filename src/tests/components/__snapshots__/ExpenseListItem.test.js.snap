// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with an expense 1`] = `
<Link
  className="list-item"
  replace={false}
  to="edit/2"
>
  <div>
    <h3
      className="list-item__title"
    >
      beer
       
    </h3>
    <span
      className="list-item__sub-title"
    >
      25 December 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $3.50
     
  </h3>
</Link>
`;
